onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -633,87 +707,80 @@ def list(self, request, *args, **kwargs):
-                    new_validation_status_uid = {}
-                else:
-                    new_validation_status_uid = payload.get("validation_status.uid")
+            cursor = ParsedInstance.query_mongo_no_paging(**query_kwargs)
+            instances_ids = [record.get('_id') for record in list(cursor)]
 
-                if new_validation_status_uid is None:
-                    http_status = status.HTTP_400_BAD_REQUEST
-                    response = {"detail": _("No validation_status.uid provided")}
-                else:
-                    # Create new validation_status object
-                    new_validation_status = get_validation_status(
-                        new_validation_status_uid, xform, request.user.username)
-
-                    # 3 scenarios to update submissions
-
-                    # First scenario / Modify submissions based on user's query
-                    if payload.get("query"):
-                        # Validate if query is valid.
-                        try:
-                            query.update(payload.get("query"))
-                        except ValueError:
-                            raise ParseError(_("Invalid query: %(query)s"
-                                               % {'query': json.dumps(payload.get("query"))}))
-
-                        query_kwargs = {
-                            "query": json.dumps(query),
-                            "fields": '["_id"]'
-                        }
-
-                        cursor = ParsedInstance.query_mongo_no_paging(**query_kwargs)
-                        submissions_ids = [record.get("_id") for record in list(cursor)]
-                        filter_.update({"id__in": submissions_ids})
-
-                    # Second scenario / Modify submissions based on list of ids
-                    elif payload.get("submissions_ids"):
-                        try:
-                            # Use int() to test if list of integers is valid.
-                            submissions_ids = payload.get("submissions_ids", [])
-                            or_ = {u"$or": [{u"_id": int(submission_id)} for submission_id in submissions_ids]}
-                            query.update(or_)
-                        except ValueError:
-                            raise ParseError(_("Invalid submissions ids: %(submissions_ids)s"
-                                               % {'submissions_ids': json.dumps(payload.get("submissions_ids"))}))
-
-                        filter_.update({"id__in": submissions_ids})
-                    # Third scenario / Modify all submissions in form, but confirmation param must be among payload
-                    elif payload.get("confirm", False) is not True:
-                        http_status = status.HTTP_400_BAD_REQUEST
-                        response = {"detail": _("No confirmations provided")}
+        # Second scenario / Get submissions based on list of ids
+        elif payload.get('submissions_ids'):
```

Comment:
`submissions_ids` sounds awkward in English to me. If we weren't naming variables, I could see a toss-up between "submissions' IDs" (possessive) and "submission IDs" (singular attributive noun), but since we can't include an apostrophe, I like `submission_ids` better. Some sentences come to mind that fit this pattern: "We need to replace all the lamp ballasts"; "Some of the fan motors have gone bad"; "The message subjects indicate a close relationship."

Reply:
Would require changes to at least:

- https://github.com/kobotoolbox/kpi/blob/502c087988acce1f72fae7a215512bd26e1a012d/jsapp/js/components/table.es6#L786-L788
- https://github.com/kobotoolbox/kpi/blob/502c087988acce1f72fae7a215512bd26e1a012d/jsapp/js/components/table.es6#L834-L835
- https://github.com/kobotoolbox/kpi/blob/502c087988acce1f72fae7a215512bd26e1a012d/kpi/views/v2/data.py#L144-L149
- https://github.com/kobotoolbox/kpi/blob/502c087988acce1f72fae7a215512bd26e1a012d/kpi/views/v1/submission.py#L143-L148

Reply:
The changes have been applied to `kpi` and I have changed `instances_ids` too.